### PR TITLE
Added SoyuzXL_Encoder

### PR DESCRIPTION
**Pull request by spacehangover**
Added SoyuzXL Encoder Macropad

## Description

Added SoyuzXL Macropad with encoder and slider support.
 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/wwa/soyuzxl_encoder/config.h
Comment (waffle87) [inline keyboards/handwired/wwa/soyuzxl_encoder/config.h:1]: This file should be removed.
- `POT1_PIN` is not used anywhere
- `ENCODERS_PAD_A/B` configuration needs to be moved to `keyboard.json`
- `LOCKING_*_ENABLE` does not make sense to enable for this keyboard given the default keymap has no modifiers

File: keyboards/handwired/wwa/soyuzxl_encoder/rules.mk
Comment (waffle87) [inline keyboards/handwired/wwa/soyuzxl_encoder/rules.mk:1]: This file should be removed. Necessary configuration needs to be moved to `keyboard.json`.

File: keyboards/handwired/wwa/soyuzxl_encoder/soyuzvia.json
Comment (waffle87) [inline keyboards/handwired/wwa/soyuzxl_encoder/soyuzvia.json:1]: This file should be removed.

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by waffle87** [comment]
Please review the PR checklist. I have unticked the box you checked asking if you have read it.

File: keyboards/handwired/wwa/soyuzxl_encoder/config.h
Comment (spacehangover) [reply keyboards/handwired/wwa/soyuzxl_encoder/config.h:1]: Hey, thanks for the review, POT1_PIN is used in the slider function.
How to i move ENCODERS_PAD_A/B to keyboard.json? i can't find any documentation on the encoder page of QMK

Comment (waffle87) [reply keyboards/handwired/wwa/soyuzxl_encoder/config.h:1]: > `POT1_PIN` is used in the slider function

Doesn't look like there's any code for this though?

As for the encoder configuration, please see [doc](https://docs.qmk.fm/reference_info_json#encoder).

Comment (spacehangover) [reply keyboards/handwired/wwa/soyuzxl_encoder/config.h:1]: Oh i see, it's ignoring the VIA keymap i was working on. Should be good now

File: keyboards/handwired/wwa/soyuzxl_encoder/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    
+}
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU),  ENCODER_CCW_CW(KC_VOLD, KC_VOLU),  ENCODER_CCW_CW(KC_VOLD, KC_VOLU),  ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+};
+#endif
+
+void matrix_scan_user(void) {
```
Comment (drashna) [inline keyboards/handwired/wwa/soyuzxl_encoder/keymaps/default/keymap.c:38]:
```suggestion
void housekeeping_task_user(void) {
```

File: keyboards/handwired/wwa/soyuzxl_encoder/keymaps/default/rules.mk [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CONSUMER_ENABLE = no
+VIA_ENABLE = yes
+ENCODER_ENABLE = yes
+ENCODER_MAP_ENABLE = yes
+MIDI_ENABLE = yes
+AUDIO_ENABLE = no
+ANALOG_ENABLE = yes
+ANALOG_DRIVER_REQUIRED = yes
```
Comment (drashna) [inline keyboards/handwired/wwa/soyuzxl_encoder/keymaps/default/rules.mk:8]:
```suggestion
ENCODER_MAP_ENABLE = yes
MIDI_ENABLE = yes
ANALOG_DRIVER_REQUIRED = yes
```

File: keyboards/handwired/wwa/soyuzxl_encoder/keyboard.json [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        "mousekey": true,
+        "nkro": true,
+        "consumer_enable": false,
+        "via_enable": true,
+        "encoder_enable": true,
+        "encoder_map_enable": true,
+        "midi_enable": true,
+        "audio_enable": false,
+        "analog_enable": true,
+        "analog_driver_required": true
```
Comment (drashna) [inline keyboards/handwired/wwa/soyuzxl_encoder/keyboard.json:21]:
```suggestion
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "encoder": true,
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/wwa/soyuzxl_encoder/config.h [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/wwa/soyuzxl_encoder/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/handwired/wwa/soyuzxl_encoder/config.h [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+
+#define POT1_PIN GP26
+
+#define ENCODER_DIRECTION_FLIP
```
Comment (drashna) [inline keyboards/handwired/wwa/soyuzxl_encoder/config.h:5]:
```suggestion
#define ENCODER_DIRECTION_FLIP

```

File: keyboards/handwired/wwa/soyuzxl_encoder/keyboard.json [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            "resolution": 4
+          },
+          {
+            "pin_a": "GP20",  
+            "pin_b": "GP21",  
+            "resolution": 4
+          }
+        ]
+      }
+}
```
Comment (drashna) [inline keyboards/handwired/wwa/soyuzxl_encoder/keyboard.json:79]:
```suggestion
}

```

File: keyboards/handwired/wwa/soyuzxl_encoder/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    static int previousPotValue = 0;
+    int currentPotValue = analogReadPin(POT1_PIN);
+    int threshold = 75;  
+
+    if (abs(currentPotValue - previousPotValue) > threshold) {
+        midi_send_cc(&midi_device, 2, 0x0A, 0x7F - (currentPotValue >> 3));
+        previousPotValue = currentPotValue;
+    }
+    
+}
```
Comment (drashna) [inline keyboards/handwired/wwa/soyuzxl_encoder/keymaps/default/keymap.c:30]: This should be moved to a `soyuzxl_encoder.c` file in the kb directory, along with the housekeeping task

File: keyboards/handwired/wwa/soyuzxl_encoder/keymaps/default/rules.mk [hunk 8]
```diff
@@ -0,0 +1,3 @@
+ENCODER_MAP_ENABLE = yes
+MIDI_ENABLE = yes
+ANALOG_DRIVER_REQUIRED = yes
```
Comment (drashna) [inline keyboards/handwired/wwa/soyuzxl_encoder/keymaps/default/rules.mk:3]: Should be moved to kb level.

**Pull request closed without merging**